contracts/pallet-ibc/src/client.rs

```diff
@@ -386,7 +376,29 @@ where
 
 		let data = consensus_state.encode_to_vec();
 		// todo: pruning
-		ConsensusStates::<T>::insert(client_id, height, data);
+		ConsensusStates::<T>::insert(client_id.clone(), height, data);
+		// We do not need this hack for neither beefy nor grandpa clients
+		if !client_id.as_str().starts_with("10-grandpa") &&
+			!client_id.as_str().starts_with("11-beefy")
+		{
+			let mut stored_heights = ConsensusHeights::<T>::get(client_id.as_bytes().to_vec());
+			if let Err(val) =
+				stored_heights.try_insert((height.revision_number, height.revision_height))
+			{
+				let first = stored_heights
+					.iter()
+					.take(1)
+					.next()
+					.expect("Cannot fail as a value always exists")
+					.clone();
+				stored_heights.remove(&first);
```

Comment:
do you need the clone above? you're using a reference afterwards

Reply:
Can't mutate the set using a reference from the set.